autoload/cmake/fileapi.vim

```diff
@@ -163,12 +161,12 @@ endf
 "     build_dir : String
 "         path to current build configuration
 "
-function! s:fileapi.Parse(build_dir) abort
+function! s:fileapi.Parse(build_dir, throw) abort
     if !l:self.cmake_version_supported
         return
     endif
 
-    let l:ret = s:ParseIndex(a:build_dir)
+    let l:ret = s:ParseIndex(a:build_dir, a:throw)
```

Comment:
I mostly agree, I don't know what I was thinking. I would throw directly from `s:ParseIndex()` however, as this allows us to throw different exception for different errors. If we still name them in a consistent way we still can catch `/vim-cmake_fileapi_.*/` in function in which we just don't care. This also allows us to return the response files array locally instead of using a 'global' variable and an error code.